### PR TITLE
Cumulus 1050 add config options

### PR DESCRIPTION
**Pull request by Jkovarik**
Update execution cleanup to allow cleanup to evaluate completed status
and 'other' status execution records separately so that non-complete
execution records may retain their payload history for a different
amount of time.

**Summary:**

Separated configuration flags for originalPayload/finalPayload cleanup such that they can be set to different retention times

Addresses [CUMULUS-1050](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1050)

## Changes
(see Summary)

## Test Plan
Things that should succeed before merging.

- [x] Unit tests
- [x] Adhoc testing
- [x] Update CHANGELOG

### CODE REVIEW

File: packages/api/config/lambdas.yml [hunk 1]
```diff
@@ -140,7 +140,10 @@ cleanExecutions:
   source: 'node_modules/@cumulus/api/dist/'
   namedLambdaDeadLetterQueue: true
   envs:
-    executionPayloadRetentionPeriod: '{{execution_payload_retention_period}}'
+    nonCompleteExecutionPayloadTimeout: '{{non_complete_execution_payload_timeout}}'
```
Comment (Jkovarik) [inline packages/api/config/lambdas.yml]: I'm mildly unhappy with either label for these configuration flags, for both typo and/or clarity reasons, but nothing better came to mind.    Please feel free to suggest something more intuitive.

File: packages/common/tests/aws.js [hunk 2]
```diff
@@ -107,7 +107,7 @@ test('pullStepFunctionEvent returns original message if message not on S3', asyn
       execution_name: 'execution'
     },
     meta: {
-      bucket: 'test bucket',
```
Comment (Jkovarik) [inline packages/common/tests/aws.js]: I'm unclear why this change was needed for eslint to pass, I didn't modify this file at all.

File: packages/api/models/executions.js [hunk 3]
```diff
@@ -76,7 +83,15 @@ class Execution extends Manager {
     const concurrencyLimit = process.env.CONCURRENCY || 10;
     const limit = pLimit(concurrencyLimit);
 
-    const updatePromises = oldExecutionRows.Items.map((row) => limit(() => this.update({ arn: row.arn }, {}, ['originalPayload', 'finalPayload'])));
+    const updatePromises = oldExecutionRows.Items.map((row) => limit(() => {
+      if (row.status === 'completed' && !disableComplete && row.updatedAt <= completeMaxMs) {
```
Comment (Jkovarik) [inline packages/api/models/executions.js]: This should get refactored into multiple DB query/filters once we're not using dynamo.   Avoiding rolling it into the query here to avoid scanning the table twice when both cleanups are both enabled.

File: packages/api/models/executions.js [hunk 4]
```diff
@@ -76,7 +83,15 @@ class Execution extends Manager {
...
+        return this.update({ arn: row.arn }, {}, ['originalPayload', 'finalPayload']);
+      }
+      if (!(row.status === 'completed') && !disableNonComplete && row.updatedAt <= nonCompleteMaxMs) {
```
Comment (ifestus) [inline packages/api/models/executions.js]: Real nit-picky but it might be quicker to short-circuit these conditionals by starting with `!disableNonComplete`

File: packages/deployment/app/config.yml [hunk 5]
```diff
@@ -4,8 +4,10 @@ default:
 
   urs_url: https://uat.urs.earthdata.nasa.gov
 
-
-  execution_payload_retention_period: 30 ## days
+  non_complete_execution_payload_timeout: 30
```
Comment (ifestus) [inline packages/deployment/app/config.yml]: It might be common sense, but I kind of liked seeing the `# days` comment. Completely up to you, though.

File: packages/api/config/lambdas.yml (hunk 1 above)
Comment (ifestus) [reply packages/api/config/lambdas.yml]: Breaking the name into parts, I think `nonComplete` and `ExecutionPayload` are good descriptors, but I'm not sure about `Timeout`. I like `RetentionPeriod`, at the risk of being too long...

`nonCompleteExecutionPayloadRetentionPeriod`, while quite long, is more accurate from my perspective.

File: packages/api/lambdas/cleanExecutions.js [hunk 6]
```diff
@@ -2,24 +2,42 @@
...
+    value: nonCompleteTimeout,
+    originalValue: process.env.nonCompleteExecutionPayloadTimeout
+  },
+  {
+    name: 'completeExecutionPayloadTimeout',
+    value: completeTimeout,
+    originalValue: process.env.completeExecutionPayloadTimeout
+  }];
+
+  if (completeDisable && nonCompleteDisable) {
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply packages/api/lambdas/cleanExecutions.js]: +1 👍

[the remark this answers is not in the document]
Comment (Jkovarik) [reply packages/api/lambdas/cleanExecutions.js]: 46de14e2

File: packages/api/config/lambdas.yml [hunk 7]
```diff
@@ -140,7 +140,10 @@ cleanExecutions:
...
+    completeExecutionPayloadTimeout: '{{complete_execution_payload_timeout}}'
+    nonCompleteExecutionPayloadTimeoutDisable: '{{non_complete_execution_payload_disable}}'
```
[the remark this answers is not in the document]
Comment (Jkovarik) [reply packages/api/config/lambdas.yml]: Yes they are.  I opted to add seperate flags rather than a magic config value, e.g. -1, etc.

File: packages/api/models/executions.js (hunk 4 above)
Comment (Jkovarik) [reply packages/api/models/executions.js]: I love constructive nit-picks, often they make the code (or me.... or both!)  better. 

In this case, 💯 .     

e7c9d8cd

File: packages/deployment/app/config.yml (hunk 5 above)
Comment (Jkovarik) [reply packages/deployment/app/config.yml]: I'm not sure config values fall under the domain of common sense.    The code itself explains what the values actually are, and the user-facing docs define what the units are, this felt like another place to miss a comment update.... that said, if you think it's useful to have there for reference, I have no strong feelings either way. 

185d5018

File: packages/api/config/lambdas.yml (hunk 1 above)
Comment (Jkovarik) [reply packages/api/config/lambdas.yml]: I was a bit less concerned about the internal reference and more so about the user facing config name.  

🤔  

The problem with nonCompleteExecutionPayloadRetentionPeriod is it tends to run fowl of our style guidelines (line length) and as such winds up being immediately re-assigned to something shorter, which may or may not be desirable.   

It also is quite a mouthful, and while descriptive feels like it might not improve readability.    Then again, the other presented options have that same issue.
